tools/repositories.bzl

```diff
@@ -14,37 +14,73 @@
 
 load(":common.bzl", "BAZEL_HASH_DICT", "BAZEL_VERSIONS")
 
-_BAZEL_BINARY_PACKAGE = "http://releases.bazel.build/{version}/release/bazel-{version}-installer-{platform}.sh"
+_BAZEL_BINARY_PACKAGE = "http://releases.bazel.build/{version}/release/bazel-{version}-without-jdk-installer-{platform}.{extension}"
 
 def _get_platform_name(rctx):
   os_name = rctx.os.name.lower()
+  
+  if os_name.startswith("mac os"):
+    return "darwin-x86_64"
+  if os_name.startswith("windows"):
+    return "windows-x86_64"                
+  
   # We default on linux-x86_64 because we only support 2 platforms
-  return "darwin-x86_64" if os_name.startswith("mac os") else "linux-x86_64"
+  return "linux-x86_64"
+
 
 def _get_installer(rctx):
   platform = _get_platform_name(rctx)
   version = rctx.attr.version
-  url = _BAZEL_BINARY_PACKAGE.format(version = version, platform = platform)
-  args = {"url": url, "type": "zip"}
+  
+  if platform.startswith("windows"):
+    extension = "exe"
+  else:
+    extension = "sh"
+
+  url = _BAZEL_BINARY_PACKAGE.format(version=version, platform=platform, extension=extension)
+  args = {"url": url, "type": "zip", "output": "bin"}
   if version in BAZEL_HASH_DICT and platform in BAZEL_HASH_DICT[version]:
     args["sha256"] = BAZEL_HASH_DICT[version][platform]
   rctx.download_and_extract(**args)
 
+
+def _extract_bazel(rctx):
+  result = rctx.execute([
```

Comment:
Extracting Bazel breaks remote execution. Any functional motivation to do this or just performance?